src/test/java/guitests/TagPersonGuiTest.java

```diff
@@ -0,0 +1,276 @@
+        aliceTagDialog.enterSearchQuery("coll").acceptSuggestedTag();
+        aliceTagDialog.close();
+
+        aliceEditDialog.pressEnter();
+
+        assertEquals("Tag: friends", alicePersonCard.getTags());
+        assertTrue(alicePersonCard.isShowingGracePeriod("Editing"));
+        sleepForGracePeriod();
+        assertEquals("Tag: friends", alicePersonCard.getTags());
+        assertFalse(alicePersonCard.isShowingGracePeriod("Editing"));
+    }
+
+    @Test
+    public void tagAndUntagPerson() {
+        PersonCardHandle alicePersonCard = personListPanel.selectCard(td.alice);
+        EditPersonDialogHandle aliceEditDialog = personListPanel.use_PERSON_EDIT_ACCELERATOR();
+        TagPersonDialogHandle aliceTagDialog = aliceEditDialog.openTagPersonDialogUsingShortcut();
+
+        aliceTagDialog.enterSearchQuery("coll").acceptSuggestedTag();
+        aliceTagDialog.close();
+
+        aliceEditDialog.pressEnter();
+
+        assertEquals("Tag: colleagues", alicePersonCard.getTags());
+        assertTrue(alicePersonCard.isShowingGracePeriod("Editing"));
+        sleepForGracePeriod();
+        assertEquals("Tag: colleagues", alicePersonCard.getTags());
+        assertFalse(alicePersonCard.isShowingGracePeriod("Editing"));
+
+        EditPersonDialogHandle aliceEditDialogTwo = personListPanel.use_PERSON_EDIT_ACCELERATOR();
+        TagPersonDialogHandle aliceTagDialogTwo = aliceEditDialogTwo.openTagPersonDialogUsingShortcut();
+
+        aliceTagDialogTwo.enterSearchQuery("coll").acceptSuggestedTag();
+        aliceTagDialogTwo.close();
+
+        aliceEditDialogTwo.pressEnter();
+
+        assertEquals("", alicePersonCard.getTags());
+        assertTrue(alicePersonCard.isShowingGracePeriod("Editing"));
+        sleepForGracePeriod();
+        assertEquals("", alicePersonCard.getTags());
+        assertFalse(alicePersonCard.isShowingGracePeriod("Editing"));
+    }
+
+    @Test
+    public void tagPersonAndChangeTag() {
+        PersonCardHandle alicePersonCard = personListPanel.selectCard(td.alice);
+        EditPersonDialogHandle aliceEditDialog = personListPanel.use_PERSON_EDIT_ACCELERATOR();
+        TagPersonDialogHandle aliceTagDialog = aliceEditDialog.openTagPersonDialogUsingShortcut();
+
+        aliceTagDialog.enterSearchQuery("coll").acceptSuggestedTag();
+        aliceTagDialog.close();
+
+        aliceEditDialog.pressEnter();
+
+        assertEquals("Tag: colleagues", alicePersonCard.getTags());
+        assertTrue(alicePersonCard.isShowingGracePeriod("Editing"));
+        sleepForGracePeriod();
+        assertEquals("Tag: colleagues", alicePersonCard.getTags());
+        assertFalse(alicePersonCard.isShowingGracePeriod("Editing"));
+
+        EditPersonDialogHandle aliceEditDialogTwo = personListPanel.use_PERSON_EDIT_ACCELERATOR();
+        TagPersonDialogHandle aliceTagDialogTwo = aliceEditDialogTwo.openTagPersonDialogUsingShortcut();
+
+        aliceTagDialogTwo.enterSearchQuery("coll").acceptSuggestedTag();
```

Comment:
The two calls are used together frequently. Can be combined to one, and chained.
`aliceTagDialog.addTag("coll").addTag("Fri")...`